cogs/help.py

```diff
@@ -1,63 +1,185 @@
+    def __init__(self):
+        super().__init__(timeout = None)
+        self._thread_author: Optional[Member] = None
+
+    # eww but need it for this view to be persistent.
+    async def get_thread_author(self, channel: Thread) -> Optional[Member]:
+        history = channel.history(oldest_first = True, limit = 1)
+        history_flat = await history.flatten()
+        user = history_flat[0].mentions[0]
+        self._thread_author = user
+
+        return self._thread_author
+
+    @ui.button(label = "Close", style = ButtonStyle.red, custom_id = "help_thread_close_button")
+    async def thread_close_button(self, button: Button, interaction: Interaction):
+        if not self._thread_author:
+            await self.get_thread_author(interaction.channel)  # type: ignore
+
+        log_channel = interaction.guild.get_channel(HELP_LOGS_CHANNEL_ID)
+        await interaction.channel.send(
+            "This thread has now been closed. Please create another thread if you wish to ask another question."
+        )
+        button.disabled = True
+        await interaction.message.edit(view = self)
+        self.stop()
+        await interaction.channel.edit(locked = True, archived = True)
+        await log_channel.send(
+            f"Help thread {interaction.channel.name} (created by {self._thread_author.name}) has been closed."
+        )
+
+    async def interaction_check(self, interaction: Interaction) -> bool:
+        thread_author = self._thread_author or await self.get_thread_author(interaction.channel)  # type: ignore
+        # because we aren't assigning the persistent view to a message_id idk.
+        if not isinstance(interaction.channel, Thread) or interaction.channel.parent_id != HELP_CHANNEL_ID:
+            return False
+
+        return interaction.user.id == thread_author.id or interaction.user.get_role(HELPER_ROLE_ID)
+
 
 class HelpCog(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
-        self.help_channel = 881965127031722004
-        self.bot.loop.create_task(self.create_view())
+        self.bot.loop.create_task(self.create_views())
 
-    async def create_view(self):
+    async def create_views(self):
         if getattr(self.bot, "help_view_set", False) is False:
             self.bot.help_view_set = True
             self.bot.add_view(HelpView())
+            self.bot.add_view(ThreadCloseView())
 
     @commands.Cog.listener()
     async def on_message(self, message):
-        if message.channel.id == self.help_channel and message.type == MessageType.thread_created:
-            await message.delete(delay=5)
+        if message.channel.id == HELP_CHANNEL_ID and message.type is MessageType.thread_created:
+            await message.delete(delay = 5)
+        if isinstance(message.channel, Thread) and \
+                message.channel.parent_id == HELP_CHANNEL_ID and \
+                message.type is MessageType.pins_add:
+            await message.delete(delay = 10)
```

Comment:
I don't see why we should delete pin messages here

Reply:
The chat looks cleaner without imo. It's after 10 seconds to the user knows that it got pinned.

Reply:
Ah fair